File: Hacktoberfest2021/QueueJava/QueueOperations.java
package QueueJava;

public class QueueOperations {
    static int front, rear, capacity, size;

    // Declaring the Queue as an array
    int Q[];

    public QueueOperations(int capacity){
        this.capacity = capacity; // Initializing th Queue with a fixed capacity
        Q = new int[capacity];
        front = this.size = 0;
        rear = capacity - 1;
    }

    // Function to check if the queue is full or not
    // Queue is full when size reaches its capacity
    public boolean isFull(){
        return Q.length == this.size;
    }

    // Function to check if Queue is completely empty or not
    // Queue is empty when size of it is 0
    public boolean isEmpty(){
        return size == 0;
    }

    // Function that returns the front element of the Queue
    public int front(){
        return Q[front];
    }

    // Function that returns the last element of the Queue
    public int rear(){
        return  Q[rear];
    }

    // Function to add an elelment in the Queue
    public void enqueue(int num){
       if(isFull()) return;
        rear = (rear + 1) % capacity;
        this.Q[rear] = num; // Adds the element at the end of the Queue
        size = size + 1;
    }

    // // Function to remove an elelment in the Queue
    public int dequeue(){
        int item = this.Q[front]; // Stores the first element to return when removed
        front = (front + 1) % capacity;
        size = size - 1; // Reduce the Queue size by 1
        return item;
    }

    // Driver method to verify the implementation of the Queue
    public static void main(String[] args) {
        QueueOperations q = new QueueOperations(3);
        System.out.println(q.isEmpty());
        System.out.println(q.isFull());
        q.enqueue(1);
        q.enqueue(2);
        q.enqueue(3);
        System.out.println(q.isFull());
        System.out.println(q.front());
        q.dequeue();
        System.out.println(q.front());
        q.dequeue();
        q.dequeue();
        System.out.println(q.isEmpty());
        System.out.println(q.isFull());

    }


}

File: Hacktoberfest2021/Queue_operations.cpp
#include <iostream>
using namespace std;
class queues
{

    int q[100]; //Queue as an array
    int f = -1; //Declaring front as -1
    int r = -1; //Declaring rear as -1
    int n;      //Number of elements to go in the queue

public:
    void enqueue(int ele); //Function to add an element into the queue
    int dequeue();         //Function to delete an element from the queue
    void display();        //For displaying the queue
    bool isfull();         //To check if the queue is full
    bool isempty();        //To check if the queue is empty
    int peek();            //To check the front element in the queue
};
int main()
{

    queues q;
    int n;
    cout << "Enter the number of elements to be inserted: ";
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        q.enqueue(n);
    }
    q.display();

    int del;
    cout << "Enter the number of elements to be deleted: ";
    cin >> del;
    for (int i = 0; i < del; i++)
    {
        q.dequeue();
    }
    q.display();
    cout << "The element at peek is: " << q.peek();
}
void queues::enqueue(int size)
{
    int ele;
    cout << "Enter the element of the queue: ";
    cin >> ele;

    if (isfull()) //i.e r == n-1
    {
        cout << "Overflow" << endl; //Overflow is the condition where the queue is already full and no more elements can be inserted
        return;
    }
    else
    {
        if (isempty()) //since we are inserting the first element f and r is still -1 , so we have to increase f once and r for each element
        {
            f = 0;
        }
        r++;
        q[r] = ele; // placing the element as r is increased for every element
    }
}
int queues::dequeue()
{
    int temp;
    if (isempty()) // i.e f and r is -1
    {
        cout << "Underflow" << endl; //Underflow is the condition where the queue is empty and no elements are left to delete from it
    }
    else
    {
        temp = q[f]; //Storing the element to be stored as deletion is only from front in queues
        if (f >= r)
        {
            f = r = -1;
        }
        else
        {
            f++; //Simply increasing front so the elements will no longer count in the queue, hence deletion from the front
        }
    }
    cout << "Deleted element is: " << temp << endl;
}
void queues::display()
{
    if (isempty())
    {
        cout << "underflow";
    }
    else
    {
        cout << "The queue is: ";
        for (int i = f; i <= r; i++)
        {
            cout << q[i] << " ";
        }
        cout << endl;
    }
}
bool queues ::isempty()
{
    if (f == -1 && r == -1)
    {
        return true;
    }
    else
        return false;
}
bool queues::isfull()
{
    if (r == n - 1) // r =n-1 means that the rear pointer will point at the last index of the taken array(i.e for size = 5 r will be 4)
    {
        return true;
    }
    else
        return false;
}
int queues::peek()
{
    return q[f];
}
//Please notw that overflow condition will never be encountered in this code because we have enqueued the elements as per the size of array.